File: tehnohem-ui/src/app/shared/services/company-service/company.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Company } from '../../model/company';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  url: string = `${environment.API_URL}/Company`;

  constructor(private http: HttpClient) { }


  getAllCustomerCompanies():Observable<Company[]>{
    return this.http.get<Company[]>(this.url+"/customers")
  }

  getAllSupplierCompanies():Observable<Company[]>{
    return this.http.get<Company[]>(this.url+"/suppliers")
  }

  getAllThirdPartyCompanies():Observable<Company[]>{
    return this.http.get<Company[]>(this.url+"/thirdPartyCompanies")
  }

  addNewCompany(newCompany : Company):Observable<any>{
    return this.http.post<any>(this.url+ "/addNewCompany" , newCompany);
  }
  
  deleteCompany(companyId: string): Observable<any>{
    return this.http.delete<any>(this.url+"/deleteCompany/"+companyId);
  }

  getAllDeliveryPlaces( companyId: string):Observable<any>{
    return this.http.get<any>(this.url+"/deliveryPlaces/"+companyId);
  }

  updateCompany(company: Company):Observable<any>{
    return this.http.post<any>(this.url+"/update", company);
  }

  getCompanyInfo(companyId : string): Observable<Company>{
    return this.http.get<Company>(this.url+"/info/"+companyId);
  }

}

File: tehnohem-api/Services/Implementation/CompanyService.cs
using tehnohem_api.DTO;
using tehnohem_api.Model;
using tehnohem_api.Services.Interface;
using tehnohem_api.UnitOfWork.Interface;

namespace tehnohem_api.Services.Implementation
{
    public class CompanyService : ICompanyService
    {
        private IUnitOfWork unitOfWork;
        public CompanyService(IUnitOfWork unitOfWork) {
            this.unitOfWork = unitOfWork;
        }

        public void addNewCompany(Company company)
        {
            this.unitOfWork.CompanyRepository.Add(company);
            this.unitOfWork.Commit();
        }

        public void deleteCompany(string companyId)
        {
            Company companyToDelete = this.unitOfWork.CompanyRepository.getById(companyId);
            this.unitOfWork.CompanyRepository.Delete(companyToDelete);
            this.unitOfWork.Commit();
        }

        public IList<Company> GetAllCompanies()
        {
            return this.unitOfWork.CompanyRepository.getAllCompanies();
        }

        public IList<Company> GetAllCustomerCompanies()
        {
            return this.unitOfWork.CompanyRepository.getAllCustomerCompanies();
        }

        public IList<Company> GetAllDeliveryPlaces(string companyId)
        {
            Company company = this.unitOfWork.CompanyRepository.getById(companyId);
            return this.unitOfWork.CompanyRepository.getAllDeliveryPlaces(company);
        }

        public IList<Company> GetAllSupplierCompanies()
        {
            return this.unitOfWork.CompanyRepository.getAllSupplierCompanies();
        }

        public IList<Company> GetAllThirdPartyCompanies()
        {
            return this.unitOfWork.CompanyRepository.getAllThirdPartyCompanies();
        }

        public Company GetCompanyById(string companyId)
        {
            return this.unitOfWork.CompanyRepository.getById(companyId);
        }

        public void updateCompany(CompanyDTO companyDTO)
        {
            Company company = this.unitOfWork.CompanyRepository.getById(companyDTO.ID);
            this.unitOfWork.CompanyRepository.Update(company,companyDTO);
            this.unitOfWork.Commit();
        }
    }
}
